kernel/src/main/jjtree/TagSelector.jjt

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2013-2020 K Team. All Rights Reserved.
+options {
+  MULTI=true;
+  KEEP_LINE_COLUMN = false;
+  STATIC = false;
+}
+PARSER_BEGIN(TagSelector)
+package org.kframework.parser.tagSelector;
+
+import java.lang.String;
+import java.io.StringReader;
+import java.util.Set;
+import java.util.HashSet;
+import org.kframework.utils.errorsystem.KEMException;
+
+/**
+ * An Arithmetic Grammar.
+ * @author Radu Mereuta
+ */
+public class TagSelector {
+
+  /** Run this to test the parser. */
+  public static void main(String args[]) {
+    System.out.println("Reading from standard input...");
+    TagSelector t = new TagSelector(new StringReader("a|!b&c"));
+    try {
+      //SimpleNode result = t.ExpressionStart();
+      SimpleNode result = t.TagListStart();
+      result.dump("");
+      System.out.println("Success.");
+    } catch (Exception e) {
+      System.out.println("Error:");
+      System.out.println(e.getMessage());
+      e.printStackTrace();
+    }
+  }
+
+  public static Set<String> parseTags(String input) {
+    TagSelector t = new TagSelector(new StringReader(input));
+    Set<String> tags = new HashSet<String>();
+    try {
+      ASTTagListStart result = t.TagListStart();
+      for (Node n : result.children)
+        if (n instanceof SimpleNode) {
+            String tag = ((SimpleNode) n).value.toString();
+            if (tag.charAt(0) == '.')
+                tag = tag.substring(1);
+            tags.add(tag);
+        }
+    } catch (Exception e) {
+      // ignore: e.printStackTrace();
+    }
+    return tags;
+  }
+
+  public static ASTExpressionStart parseSelectorExp(String input) {
+    TagSelector t = new TagSelector(new StringReader(input));
+    try {
+      ASTExpressionStart result = t.ExpressionStart();
+      return result;
+    } catch (ParseException e) {
+      throw KEMException.compilerError("Could not parse " + input + " as a selector expression.", e);
```

Comment:
This needs better error message. Take a look at how we format the error messages from javacc in the outer parser.